src/clr-angular/data/datagrid/datagrid-if-expanded.service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+
+import { IfExpandService } from '../../utils/conditional/if-expanded.service';
+import { ClrLoadingState } from '../../utils/loading/loading';
+
+@Injectable()
+export class DatagridIfExpandService extends IfExpandService {
+  // due to the es5 spec if the set is overridden on base class the getter must also be overridden
+  get expanded(): boolean {
+    return this._expanded;
+  }
+
+  set expanded(value: boolean) {
```

Comment:
This setter seems to duplicate the base class's setter. Can we directly subscribe to `expandChange` in the `row-expand-animation.ts` so we don't even need to create another subject for animation only?

Reply:
Hmm I don't think so, the issue I ran into was that the animation observable needs to fire when `loadingStateChange()` is called to know if it is not loading currently so I couldn't get rid of it. Once the row if updated to use ng animations this should be able to go away altogether which is what I think @Jinnie was referring to.

Reply:
ah I see.